src/externalTypes.ts

```diff
@@ -0,0 +1,29 @@
+interface SharedOpts {
+  /** JS target */
+  target: 'node' | 'browser';
+  /** Path to tsconfig file */
+  tsconfig?: string;
+  /** Is error extraction running? */
+  extractErrors?: boolean;
+}
+
+type ModuleFormat = 'cjs' | 'umd' | 'esm' | 'system';
+
+export interface TsdxOptions extends SharedOpts {
+  /** Name of package */
+  name: string;
+  /** path to file */
+  input: string;
+  /** Environment */
+  env: 'development' | 'production';
+  /** Module format */
+  format: ModuleFormat;
+  /** Is minifying? */
+  minify?: boolean;
+  /** Is this the very first rollup config (and thus should one-off metadata be extracted)? */
+  writeMeta?: boolean;
+  /** Only transpile, do not type check (makes compilation faster) */
+  transpileOnly?: boolean;
+}
+
+export { RollupOptions } from 'rollup';
```

Comment:
I was thinking of using the _exact_ type that TSDX modifies this too, although that may not be possible to override

Reply:
Nevermind, TSDX itself uses `RollupOptions` internally.

Reply:
So I ran `tsc` over a `tsdx.config.js` with the proper import and a `// @ts-check` and `allowJs: "true"` etc and that actually failed because of the lack of exact typing. E.g.:

```bash
$ tsc --noEmit
tsdx.config.js:13:5 - error TS2532: Object is possibly 'undefined'.

13     config.plugins.push(
```

Now the problem is that I'm not entirely sure how to do this in TS or if it's possible to. TSDX's Rollup config should _conform_ to `RollupOptions` but should also be more specific than `RollupOptions` -- e.g. we should _know_ that `config.plugins.push` is possible without any type-casting.

Of course, TSDX Rollup config could have its own interface that _extends_ `RollupOptions`, but that would be the naive route and could still cause users some type-checking problems. The config is complex enough that I'd like it to be inferred (especially the plugins' types, since users may override them) while still conforming to `RollupOptions` (so we don't get problems like #371 ). I'm not sure if that's possible in TS or if I just haven't been able to figure out how to do it in TS.